go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/reddit/baseplate.go
 
-go 1.22.0
+go 1.23
```

Comment:
🔕 I think it wants 1.23.0 nowadays... I've run into a few issues with totally dropping the third dot, e.g. when things parse it out and expect e.g. `GOTOOLCHAIN=<that>` to work

Reply:
do you have an example of dropping patch release from `go` line breaks things?

Reply:
will merge this as-is. we can add `.0` if this really breaks something (I don't think it will)